add_item_form.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+$db = new PDO ('mysql:host=db;dbname=collection', 'root', 'password');
+
+$db -> setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+$nameQuery = $db->prepare( 'SELECT `name` FROM `games`;');
+
+$nameQuery->execute();
+
+$names = $nameQuery -> fetchAll();
+
+
+class Game
+{
+    protected $name;
+    protected $genre;
+    protected $length;
+    protected $price;
+
+    /**
+     * The constructor for the Game object, only needs a name and the db to search the db and create a new object, if the add section is set to true th constructor can add a new item to the db
+     *
+     * @param String $name
+     * @param PDO $db
+     * @param bool $add
+     * @param String $genre
+     * @param String $length
+     * @param String $price
+     */
+    public function __construct(String $name, PDO $db, Bool $add = false, String $genre = '', String $length ='', String $price ='')
+    {
+        if($this->nameSearch($name,$db) === true){
+            $this -> name = $name;
+            $this -> getStats($name,$db);
+        }
+        elseif($add === true){
+            $this->addStats($name, $genre, $length, $price, $db);
+        }
+        else{
+            echo 'The name you entered is not in the collection.';
+        }
+
+    }
+
+    /**
+     * returns an array of all the objects parameters.
+     *
+     * @return array
+     */
+    public function getParams() :Array
+    {
+        return ['name' => $this -> name, 'genre' =>$this -> genre, 'length' => $this -> length, 'price' => $this -> price];
+    }
+    
+    /**
+     * Checks to see if the inputted name is in the db.
+     *
+     * @param String $name
+     * @param PDO $db
+     * @return bool
+     */
+    protected function nameSearch( String $name, PDO $db) : Bool
+    {
+        $nameQuery = $db->prepare( 'SELECT `name` FROM `games`;');
```

Comment:
Didn't you do this query at the start? Can you add this query into a function at the start and feed the result into here?

Reply:
changed the object to just pull form the full db queried at the start